files-community/Files
# Feature: Git Integration Phase 3

**Resolved / Related Issues**
   Related #11014 

**IMPORTANT**
Please, if you try checking out remotes, place a breakpoint at line 149 of GitHelpers.cs and make sure that the method is executed only once
There was a bug where it would keep on creating new branches because of selection changing. It should be fixed now, but it's better to be careful 

**Validation**
How did you test these changes?
   1. Open app and navigate to a git directory
   2. Click on the branch name
   3. Click on `New Branch`

**Screenshots (optional)**
![2](https://github.com/files-community/Files/assets/102259289/6658867f-40db-457e-81bc-1ed53e141aac)
